pkg/maps/ctmap/ctmap.go

```diff
@@ -344,6 +370,10 @@ func doGC4(m *bpf.Map, filter *GCFilter) int {
 // It returns how many items were deleted from m.
 func GC(m *bpf.Map, mapName string, filter *GCFilter) int {
 	if filter.fType&GCFilterByTime != 0 {
+		// If LRUHashtable, no need to garbage collect as LRUHashtable cleans itself up.
```

Comment:
What is the reason for only doing this check for LRUHashtable if `filter.fType&GCFilterByTime != 0` as opposed to outside of this function regardless of the filtering mechanisms which is currently done in master in `RunGC`?

Reply:
I think this is a matter of whether we think it is the `pkg/endpointmanager`'s job to handle differences between LRUHashTable or other types, or if it's `pkg/ctmap`'s job to do this. I think it should be the latter.

Reply:
@ianvernon @joestringer the real reason is that the GC does not need to be executed when using a filter type by time if the map type LRUHashtable as it cleans itself up over time.